quickcheck-contractmodel/src/Test/QuickCheck/ContractModel/Internal.hs

```diff
@@ -108,8 +109,8 @@ instance (Monad m, HasChainIndex m) => HasChainIndex (RunMonad m) where
   getChainState = lift getChainState
 
 instance (Monad m, HasChainIndex m) => HasChainIndex (StateModel.PostconditionM m) where
-  getChainIndex = lift getChainIndex
-  getChainState = lift getChainState
+  getChainIndex = StateModel.PostconditionM (lift getChainIndex)
```

Comment:
@sjoerdvisscher why is this necessary?

Reply:
The current version of `quickcheck-dynamic` on Hackage doesn't have this instance. It's now fixed on `main` but it is not on Hackage yet.

Reply:
🤦 of course... I remember when that happened... We should probably fix that over in `quickcheck-dynamic` soon...